packages/block-library/src/site-logo/edit.js

```diff
@@ -627,8 +627,7 @@ export default function LogoEdit( {
 								render={ ( { open } ) => (
 									<div className="block-library-site-logo__inspector-upload-container">
 										<Button
-											// TODO: Switch to `true` (40px size) if possible
-											__next40pxDefaultSize={ false }
+											__next40pxDefaultSize
```

Comment:
This can be seen, while you add `core/site-logo` block, and check the inspector controls to upload the logo.

**Note:** This has some styles that have higher specificity and specific to this button on this block, hence before and after is same, but we can see our change is applied in inspector controls.

| Before | After     |
| :-------- | :------- |
| <img width="576" alt="site-logo-before" src="https://github.com/user-attachments/assets/502611a9-614d-4241-ae1b-0c1a92ad9691"> | <img width="576" alt="site-logo-after" src="https://github.com/user-attachments/assets/3c3517f3-5664-48f4-8574-e5c0649de1e3"> |

Reply:
We have [style overrides](https://github.com/WordPress/gutenberg/blob/ff01bcbd4aeb84d6a28a655ca652a9ad9f16cebe/packages/block-library/src/site-logo/editor.scss#L120-L147) in this case as well.

For the context of this PR, I think the current code changes are enough.

---

Still, I'd love to remove all of those overrides if possible, and just use a standard `Button` variant — although the situation is trickier than it looks. There are three possible UIs shown to the user:

- [an upload button](https://github.com/WordPress/gutenberg/blob/f5ddf47f0d51b36c5179926efa338356f547d6f9/packages/block-library/src/site-logo/edit.js#L622-L645) (the one in the image above)
- a ["readonly" item group](https://github.com/WordPress/gutenberg/blob/f5ddf47f0d51b36c5179926efa338356f547d6f9/packages/block-library/src/site-logo/edit.js#L601-L610) showing the logo preview;
- an [interactive dropdown](https://github.com/WordPress/gutenberg/blob/f5ddf47f0d51b36c5179926efa338356f547d6f9/packages/block-library/src/site-logo/edit.js#L611-L621) allowing to replace/remove the current logo

Therefore, we'd need to make sure that we could offer a good, consistent UI in all three cases even without those overrides. Something we could do in a follow-up (cc @WordPress/gutenberg-design )/

Also, I noticed that the dropdown comes from the [`MediaReplaceFlow` component](https://github.com/WordPress/gutenberg/blob/aa2e342480b5eb7961165fb2a1b8979c65e68e43/packages/block-editor/src/components/media-replace-flow/index.js). That dropdown is using a `ToolbarButton` as its trigger, showing that it was only meant to be rendered in a toolbar — while here it's being used in the editor sidebar (cc @WordPress/gutenberg-components  @ntsekouras @youknowriad ).